### DISCUSSION TITLE
Compiler errors when installing treesitter through my init.lua

### DISCUSSION THREAD

**Discussion by frankieD3** [nvim-treesitter/nvim-treesitter#6827] [Q&A]
I have added treesitter to my init.lua and to be managed by lazy. I get errors as it tries to compile the treesitter code locally. I suspect the problem  is the compiler and standard used. I have tried some solutions I found on line by they all place a treesitter function that hasn't been installed yet.

nvim-treesitter[cpp]: Error during compilation
src/scanner.c: In function 'tree_sitter_cpp_external_scanner_serialize':^M
src/scanner.c:126:5: warning: implicit declaration of function 'static_assert'
 [-Wimplicit-function-declaration]^M
     static_assert(MAX_DELIMITER_LENGTH * sizeof(wchar_t) < TREE_SITTER_SERIAL
IZATION_BUFFER_SIZE,^M
     ^~~~~~~~~~~~~^M
C:\Users\fdemi\AppData\Local\Temp\ccPVKkQb.o:scanner.c:(.text+0x199): undefine
d reference to `static_assert'^M
[init.txt](https://github.com/user-attachments/files/16033961/init.txt)

collect2.exe: error: ld returned 1 exit status^M

My init.lua is in the attachment.
[init.txt](https://github.com/user-attachments/files/16033968/init.txt)

Any ideas?
Thanks,
Frank @

**Comment by clason** [comment]
We can't help with lazy; you need to make sure a compatible compiler is installed and on path, see https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support.

**Reply by frankieD3** [reply]
This isn't a lazy issue. Somewhere in the process of initialization a compiler is invoked to compile your codes . How do I find out which one. 
Another thing why does this site not accept .lua files as an attachment.

Yes  I am a beginner at Neo but have used vi for nearly 40 years.

Thanks,
Frank

**Reply by frankieD3** [reply]
Another point how does one use
require 'nvim-treesitter.install'.compilers = { "clang" }

if treesitter won't install?
Frank
